index.bs

```diff
@@ -5681,6 +5692,313 @@ The [=remote end subscribe steps=], with [=subscribe priority=] 10, given
 
 </div>
 
+## Input ## {#module-input}
+
+The <dfn export for=modules>input</dfn> module contains functionality for
+simulated user input.
+
+### Definition ### {#module-input-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+InputCommand = (
+  input.PerformActions //
+  input.ReleaseActions
+)
+</pre>
+
+### Types ### {#module-input-types}
+
+#### input.ElementOrigin #### {#type-input-origin}
+
+The <code>input.ElementOrigin</code> type represents an {{Element}} that will
+be used as a coordinate origin.
+
+<pre class="cddl remote-cddl">
+input.ElementOrigin = {
+  type: "element",
+  element: SharedReference
+}
+</pre>
+
+<div algorithm>
+The <dfn>is <code>input.ElementOrigin</code></dfn> steps given |object| are:
+
+1. If |object| is a [=/map=] matching the <code>input.ElementOrigin</code>
+   production, return true.
+
+1. Return false.
+
+</div>
+
+<div algorithm>To <dfn>get Element from <code>input.ElementOrigin</code>
+steps</dfn> given |session|:
+
+1. Return the following steps, given |origin| and |context|:
+
+  1. Assert: |origin| matches <code>input.ElementOrigin</code>.
+
+  <!-- converting to realm here is kind of silly since we immediately convert back
+       in [=deserialize remote reference=] -->
+  1. Let |document| be |context|'s [=active document=].
+
+  1. Let |reference| be |origin|["<code>element</code>"]
+
+  1. Let |environment settings| be the [=environment settings object=] whose
+     [=relevant global object=]'s <a>associated <code>Document</code></a> is
+     |document|.
+
+  1. Let |realm| be |environment settings|' [=realm execution context=]'s
+     Realm component.
+
+  1. Let |element| be the result of [=trying=] to [=deserialize remote reference=]
+     with |reference|, |realm|, and |session|.
```

Comment:
Note that this addition will conflict with the changes on https://github.com/w3c/webdriver-bidi/pull/385 so please do not forget to update whether which of the PRs are merged first.

Reply:
I was actually wrong. In PR #385 we only made modifications to the serialization part and that is not affected here.